CI driver_weekly bugfix deleting new branch if is already on repo

TerrenceMcGuinness-NOAA opened the pull request:
# Description
Bugfix and updates to weekly CI test that deletes the new modfied branch  if it is already on the repo but now also makes sure that the intermediary branch that creates the PR is coming from **emcbot**.

# Type of chang
- Bugfix (CI weekly test can fail if ran twice)
- Does not push branch to NOAA-EMC global-workflow branch but comes from emcbot instead

# How has this been tested?
- Ran the script in cron on Orion (didn't work on Hera because gh is to old there)

TerrenceMcGuinness-NOAA committed:
driver_weekly bugfix deleting new branch if is already on repo

TerrenceMcGuinness-NOAA committed:
Update driver_weekly.sh

remove testing directly on $?

aerorahul commented on ci/scripts/driver_weekly.sh:
```diff
@@ -22,7 +22,7 @@ set -eux
 #################################################################
 export GH=${HOME}/bin/gh
-export REPO_URL=${REPO_URL:-"https://github.com/NOAA-EMC/global-workflow.git"}
+export REPO_URL="ssh://git@ssh.github.com:443/NOAA-EMC/global-workflow.git"
```
Can I ask why `ssh`?

TerrenceMcGuinness-NOAA replied:
more secure since this checkout needs to push the changes to the EMC Reop with the updated cases directory  ..  also this change is more about **not** getting REPO from env.

WalterKolczynski-NOAA replied:
I'm still nervous about making/deleting branches in the auth repo. But I agree to not having these in a personal account either. Can emc-bot make a clone of global-workflow where the temp branches could be made?

TerrenceMcGuinness-NOAA replied:
Yes it was doing that before during development.  I will make this change in this PR.

aerorahul reviewed (commented):
Looks good other than the `ssh` bit.

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into CI-bugfix-driver_weekly

TerrenceMcGuinness-NOAA committed:
changed REPO_OWNER away from NOAA-EMC so that intermediary branch does not get pushed to authoritative repo

WalterKolczynski-NOAA commented on ci/scripts/driver_weekly.sh:
```diff
@@ -84,7 +90,7 @@ git push --set-upstream origin "${branch}"
 ####################################################################
 
-REPO_OWNER="NOAA-EMC"
+REPO_OWNER="TerrenceMcGuinness-NOAA"
```
Can we use emc-bot for this?

TerrenceMcGuinness-NOAA replied:
Yes updated the script so it now uses **emcbot**

TerrenceMcGuinness-NOAA committed:
updated driver_weely so that intermediary branch is stored in emcbot

WalterKolczynski-NOAA commented on ci/scripts/driver_weekly.sh:
```diff
@@ -73,26 +73,38 @@ git checkout -b "${branch}"
 git commit -m "Moved weekly cases files into pr for high resolution testing"
-git push --set-upstream origin "${branch}"
+
+git remote add upstream "git@github.com:${REPO_OWNER}/${REPO_NAME}.git"
```
Recommend using `emcbot` (or maybe just `${REPO_NAME}`) instead of `upstream` for the remote name to avoid confusion since "upsteam" already has a meaning in git parlance.

TerrenceMcGuinness-NOAA replied:
yes will do ...

WalterKolczynski-NOAA reviewed (changes requested):
Is there

WalterKolczynski-NOAA commented:
Other than the one quibble about the remote name, I think this is ready if it works (I see Orion failed, but not sure if it is related to these changes).

TerrenceMcGuinness-NOAA committed:
micro update for remote name for pushing weekly update brandh to emcbot repo

TerrenceMcGuinness-NOAA committed:
picked up two other name chages in remote stream for pushing the emcbot repo

TerrenceMcGuinness-NOAA commented on ci/scripts/driver_weekly.sh:
```diff
@@ -73,26 +73,39 @@
+git ls-remote --exit-code upstream "${branch}"
+ci_status=$?
+if [[ "${ci_status}" == '0' ]]; then
+    git push "${REMMOT_NAME}" --delete "${branch}"
```
oh rats!!!

TerrenceMcGuinness-NOAA committed:
rats -- spelled REMOTE worng .. fixed

TerrenceMcGuinness-NOAA reviewed (commented):
fixed and double fixed REMOTE_REPO name changes

TerrenceMcGuinness-NOAA commented:
> Other than the one quibble about the remote name, I think this is ready if it works (I see Orion failed, but not sure if it is related to these changes).

Yeah they can't because this script doesn't actually run the test, the test run itself is entirely separate.

TerrenceMcGuinness-NOAA reviewed (commented):
done

TerrenceMcGuinness-NOAA committed:
Update driver_weekly.sh

Removed "High Resolution" from title to short the name to prevent the unsightly wrapping of the labels.

TerrenceMcGuinness-NOAA committed:
Update driver_weekly.sh

still had REMOTE_NAME not updated in checking if its already there
